packages/react-hot-loader/test/AppContainer.dev.test.js

```diff
@@ -1340,7 +1340,7 @@ describe(`AppContainer (dev)`, () => {
 
         // How it works. IndeterminateComponent calculates return value once.
         expect(wrapper.text()).toBe('hey 44 new')
-        expect(spy).toHaveBeenCalledTimes(1) // called during reconciler, has no effect on result
+        expect(spy).toHaveBeenCalledTimes(0) // never gets called
```

Comment:
Not it mimics React behaviour better.

Reply:
yeah... it's just a laziness thing. i think one of the old versions of RHL had a proxy strategy that worked on instances (maybe?), but it's not really worth it since relay queries are static anyway, and nobody else uses this pattern

Reply:
Why somebody even invents it!? 🔮

Reply:
I believe it’s because Relay containers are relatively expensive to create, and idiomatic Relay code puts a container around nearly every component. Given that, my guess is that it’s an optimization to improve startup time. 

I think that’s also why nobody else uses this pattern, and why we don’t really have to worry about it. Unless you’re building static container classes everywhere, there’s no point to using it.

Reply:
A container class around every component class, I mean. In Relay, nearly every component is a smart component, sort of.

Reply:
Can this be merged and released then?

Reply:
Note that this is against the `indeterminateComponent` feature branch, not `next`.

Until this gets released, you can still use this monkeypatch, though you'd need to do a bit more complicated stuff with modules for modern/compat than for classic/legacy: https://github.com/gaearon/react-hot-loader/issues/775#issuecomment-356508850.